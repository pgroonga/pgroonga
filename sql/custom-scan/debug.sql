/* Temporary test to be used during implementation of a custom scan. */

CREATE TABLE memos (content text);
CREATE INDEX memos_content ON memos USING pgroonga (content);
INSERT INTO memos VALUES ('PostgreSQL is a RDBMS.');
INSERT INTO memos VALUES ('Groonga is fast full text search engine.');
INSERT INTO memos VALUES ('PGroonga is a PostgreSQL extension that uses Groonga.');

SET pgroonga.enable_custom_scan = on;

EXPLAIN (COSTS OFF)
SELECT content, pgroonga_score(tableoid, ctid)
  FROM memos
 WHERE content &@~ 'PGroonga OR Groonga';

SELECT content, pgroonga_score(tableoid, ctid)
  FROM memos
 WHERE content &@~ 'PGroonga OR Groonga';

DROP TABLE memos;
