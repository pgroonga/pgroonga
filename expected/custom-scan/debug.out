/* Temporary test to be used during implementation of a custom scan. */
CREATE TABLE memos (content text);
CREATE INDEX memos_content ON memos USING pgroonga (content);
INSERT INTO memos VALUES ('PostgreSQL is a RDBMS.');
INSERT INTO memos VALUES ('Groonga is fast full text search engine.');
INSERT INTO memos VALUES ('PGroonga is a PostgreSQL extension that uses Groonga.');
SET pgroonga.enable_custom_scan = on;
EXPLAIN (COSTS OFF)
SELECT content
  FROM memos
 WHERE content &@~ 'PGroonga OR Groonga';
                     QUERY PLAN                      
-----------------------------------------------------
 Custom Scan (PGroongaScan) on memos
   Filter: (content &@~ 'PGroonga OR Groonga'::text)
   PGroongaScan: DEBUG
(3 rows)

SELECT content
  FROM memos
 WHERE content &@~ 'PGroonga OR Groonga';
                        content                        
-------------------------------------------------------
 PostgreSQL is a RDBMS.
 Groonga is fast full text search engine.
 PGroonga is a PostgreSQL extension that uses Groonga.
(3 rows)

DROP TABLE memos;
